src/firebolt/utils/util.py

```diff
@@ -181,3 +190,21 @@ def __exit__(
         ):
             log_message = self._message + self.elapsed_time + "s"
             logger.debug(log_message)
+
+
+def parse_url_and_params(url: str) -> Tuple[str, Dict[str, str]]:
+    """Extract URL and query parameters separately from a URL."""
+    url = fix_url_schema(url)
+    parsed_url = urlparse(url)
+    query_params = parse_qs(parsed_url.query)
+    # This strips query parameters from the URL
```

Comment:
nit: Extend this comment a little. I was confused from the `urljoin` functionality first before I read the docs.